### PR TITLE
Refactor JS - Array & Hash

### PR DESCRIPTION
**Pull request by aakhtar3**

### CODE REVIEW

File: javascript/695-max-area-of-island.js [hunk 1]
```diff
@@ -2,32 +2,35 @@
  * @param {number[][]} grid
  * @return {number}
  */
- var maxAreaOfIsland = function(grid) {
```
Comment (aakhtar3) [inline javascript/695-max-area-of-island.js:5]: Git doesn't like two files that have a diff with same name, but different case styling
https://github.com/neetcode-gh/leetcode/blob/main/javascript/695-Max-Area-Of-Island.js
https://github.com/neetcode-gh/leetcode/blob/main/javascript/695-max-area-of-island.js

File: javascript/347-Top-K-Frquent-Elements.js [hunk 2]
```diff
@@ -1,30 +1,97 @@
 /**
+ * https://leetcode.com/problems/top-k-frequent-elements/
```
Comment (aakhtar3) [inline javascript/347-Top-K-Frquent-Elements.js:2]: Delete file due to typo?

**Comment by aakhtar3** [comment]
@mitchellirvin Added detailed complexities with multi solutions. Is this format helpful?

File: javascript/347-Top-K-Frequent-Elements.js [hunk 3]
```diff
@@ -1,30 +1,97 @@
 /**
+ * https://leetcode.com/problems/top-k-frequent-elements/
+ * Time O(N^2) | Space O(N^2)
```
Comment (aakhtar3) [inline javascript/347-Top-K-Frequent-Elements.js:3]: Verify

File: javascript/271-Encode-and-Decode-Strings.js [hunk 4]
```diff
@@ -1,29 +1,40 @@
 /**
+ * https://leetcode.com/problems/encode-and-decode-strings/
+ * Time O(N) | Space O (N)
```
Comment (aakhtar3) [inline javascript/271-Encode-and-Decode-Strings.js:3]: Add bit

**Comment by mitchellirvin** [comment]
Sorry @aakhtar3 ! Been so busy. Can review this if you don’t mind resolving the conflicts?

**Comment by aakhtar3** [comment]
@mitchellirvin I updated the PR and resolved conflicts.

Any update on helping this repo from our [last chat](https://github.com/neetcode-gh/leetcode/pull/464)?

I also have a few other PR that went stale and pending, see my [PRs](https://github.com/neetcode-gh/leetcode/pulls?q=+is%3Apr+author%3Aaakhtar3+)

File: javascript/217-Contains-Duplicate.js [hunk 5]
```diff
@@ -1,31 +1,67 @@
...
  */
-
-//First method using Set() (exit early if true)
-var containsDuplicate = function (nums) {
-    const numsSet = new Set();
-    for (const i of nums) {
-        if (numsSet.has(i)) {
-            return true;
+ var containsDuplicate = function (nums) {
+    for (let i å= 0; i < nums.length; i++) {/* Time O(N) */
```
Comment (aakhtar3) [inline javascript/217-Contains-Duplicate.js:9]: Typo

**Review by aakhtar3 — COMMENTED** [review]
Fix typo

**Comment by mitchellirvin** [comment]
> @mitchellirvin Added detailed complexities with multi solutions. Is this format helpful?

This is terrific.

**Pull request merged**